### PR TITLE
Memoize transaction size and hash at the same time

### PR DESCRIPTION
**Pull request by fab-10**
Signed-off-by: Fabio Di Fabio <fabio.difabio@consensys.net>

## PR description

Since to calculate the transaction hash and size, the transaction needs to be encoded to bytes, we can cache both values at the same time, in order to avoid extra expensive encoding tasks.

## Fixed Issue(s)

relates to  #4724

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionEncoder.java [hunk 1]
```diff
@@ -49,10 +49,16 @@ public static void encodeForWire(final Transaction transaction, final RLPOutput
...
+
+  public static void encodeForWire(
+      final TransactionType transactionType, final Bytes opaqueBytes, final RLPOutput rlpOutput) {
+    checkNotNull(transactionType, "Transaction type was not specified.");
     if (TransactionType.FRONTIER.equals(transactionType)) {
-      encodeFrontier(transaction, rlpOutput);
+      rlpOutput.writeRaw(opaqueBytes);
     } else {
-      rlpOutput.writeBytes(encodeOpaqueBytes(transaction));
+      rlpOutput.writeBytes(opaqueBytes);
```
Comment (garyschulte) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionEncoder.java:61]: I scrutinized this to be sure we are not creating a regression for non-frontier transactions.  LGTM

**Review by garyschulte — APPROVED** [review]
:ship:

**Comment by garyschulte** [comment]
looks like there is a build failure for NewPooledTransactionHashesMessageProcessorTest  since cacluateSize() is gone now.

**Comment by fab-10** [comment]
> looks like there is a build failure for NewPooledTransactionHashesMessageProcessorTest since cacluateSize() is gone now.

Fixed

**Review by Gabriel-Trintinalia — APPROVED** [review]
LGTM

**Pull request merged**